### PR TITLE
238 bring back internal examples

### LINKED ISSUE #238: Move internal examples from `dev/` folder into regular `R/*.R` files and use `getFromNamespace()` instead of `:::`
This file https://github.com/insightsengineering/teal.reporter/blob/main/dev/internal_examples.R was created out of multiple examples of internal functions that used `:::`. The usage of `:::` in CRAN is highly discouraged. Since we wanted to keep those examples for future developers, but not necessarily pack them in the package we moved them to `dev/` folder. `dev/` folder is listed in `.Rbuildignore` and is not included in R package during the build.

After an internal discussion about the same manner for `teal.code` package we decided to keep internal examples in their respective `R/*.R` files. However we decided to substitute `:::` with `getFromNamespace()` function.

### PR DESCRIPTION
**Pull request by m7pr**
Close #238 

Examples moved to `dev/` folder during https://github.com/insightsengineering/teal.reporter/pull/209 were brough back to their places. We substituted `:::` usage, that is discouraged by CRAN, with `getFromNamespace()` function

### CODE REVIEW

File: R/Archiver.R [hunk 1]
```diff
@@ -6,6 +6,8 @@ Archiver <- R6::R6Class( # nolint: object_name_linter.
     #' @description Returns an `Archiver` object.
     #'
     #' @return an `Archiver` object
+    #' @examples
+    #' archiver <- getFromNamespace("Archiver", "teal.reporter")$new()
```
Comment (chlebowa) [inline R/Archiver.R:10]: Please import the function/class and use imported.
```suggestion
    #' Archiver <- getFromNamespace("Archiver", "teal.reporter")
    #' Archiver$new()
```

**Review by chlebowa — CHANGES_REQUESTED** [review]
I'm not sure if `$initialize` methods that take no arguments really require examples.

File: R/Archiver.R (hunk 1 above)
Comment (m7pr) [reply R/Archiver.R:10]: alrighty, that will take some time but is actually more visible for the end user on what's goin on : )
Will do one task first in here https://github.com/insightsengineering/teal.data/issues/238 and then get back to this one

Comment (m7pr) [reply R/Archiver.R]: Changed this in 5d000820789beec4a2359841f148f8b9b3e0bb2b for other examples as well

File: R/Archiver.R [hunk 2]
```diff
@@ -6,6 +6,9 @@ Archiver <- R6::R6Class( # nolint: object_name_linter.
     #' @description Returns an `Archiver` object.
     #'
     #' @return an `Archiver` object
+    #' @examples
+    #' archiver <- getFromNamespace("Archiver", "teal.reporter")
```
Comment (chlebowa) [inline R/Archiver.R:10]: Maintain class name when importing.
```suggestion
    #' Archiver <- getFromNamespace("Archiver", "teal.reporter")
```

**Review by chlebowa — CHANGES_REQUESTED** [review]

File: R/Archiver.R (hunk 2 above)
Comment (m7pr) [reply R/Archiver.R:10]: @chlebowa unsure if our lintr job will not update it afterwards to lowercase but will try

Comment (chlebowa) [reply R/Archiver.R]: If linter complains, kill the linter. The examples must present the object in its natural form.

Comment (m7pr) [reply R/Archiver.R]: ok it didnt : ) will change where needed

Comment (m7pr) [reply R/Archiver.R]: Hey, incorporated this update. I think it's way easier to track if we use class names. Thanks

**Review by chlebowa — APPROVED** [review]

**Pull request merged**